File: Java/ScheduledService.java
import java.util.concurrent.Executors;
import java.util.concurrent.ScheduledExecutorService;
import java.util.concurrent.TimeUnit;

class Service extends Thread {

    private Thread t;
    private String threadName;

    /*
     * 0 is Timer
     * 1 is CreateData
     * 2 is CreateThenDeleteData
     * 3 is CreateAndDeleteNodes
     * 4 is CreateAndDeleteLists
     * 5 is CreateAndDeleteCycles
    */
    private int serviceType;

    Service(String threadName, int serviceType){

        this.threadName = threadName;
        this.serviceType = serviceType;

        System.out.println("Creating " + threadName);

    }

    public void run(){

        System.out.println("Running " + threadName + " for " + serviceType);
        // try {
            if(serviceType == 0){
                ScheduledExecutorService execService = Executors.newScheduledThreadPool(5);
                execService.scheduleAtFixedRate(()->{
                    System.out.println(System.currentTimeMillis());
                    }, 0, 10L, TimeUnit.MILLISECONDS);
            } else if(serviceType == 1){
                CreateData cd = new CreateData();
                cd.createData(100000, 1);
                System.out.println("Create Data test finished.");
            } else if(serviceType == 2){
                CreateThenDeleteData ctdd = new CreateThenDeleteData();
                ctdd.createThenDeleteData(100000, 1);
                ctdd.createThenDeleteAllData(100000, 1);
                System.out.println("Create then Delete Data test finished.");
            } else if(serviceType == 3){
                CreateAndDeleteNodes cadn = new CreateAndDeleteNodes();
                cadn.createAndDeleteNodes(10000, 100000, 1, 2);
                System.out.println("Create and Delete Nodes test finished.");
            } else if(serviceType == 4){
                CreateAndDeleteLists cadl = new CreateAndDeleteLists();
                cadl.createAndDeleteLists(10000, 100, 100000, 1, 2);
                System.out.println("Create and Delete Lists test finished.");
            } else if(serviceType == 5){
                CreateAndDeleteCycles cadc = new CreateAndDeleteCycles();
                cadc.createAndDeleteCycles(10000, 100, 100000, 1, 2);
                System.out.println("Create and Delete Cycles test finished.");
            } else {
                System.out.println(serviceType + " is not a valid service type input");
            }
            // throw new MyInterruptedException("MyInterruptedException.");
        // } catch (MyInterruptedException e){
        //     System.out.println("Thread " + threadName + " interrupted.");
        // }
        System.out.println("Thread " + threadName + " exiting.");

    }

    public void start(){
        System.out.println("Starting " + threadName);
        if(t == null){
            t = new Thread(this, threadName);
            t.start();
        }
    }

}

public class ScheduledService {

    public static void main(String args[]){

        Service timer = new Service("Timer-thread", 0);
        Service data = new Service("Data-thread", 4); // also test for 1-5

        timer.start();
        data.start();

    }

}

File: Python/ScheduledService.py
import threading
import time

exitFlag = 0

class ScheduledService (threading.Thread):
    def __init__(self, threadID, testType, delay):
        threading.Thread.__init__(self)
        self.threadID = threadID
        self.testType = testType
        self.delay = delay
    def run(self):
        print "Starting " + self.name
        test(self.testType, self.delay)
        print "Exiting " + self.name

def test(testType, delay):
    if testType == "Timer":
        for x in range(0, 500): # 100 = 100*10 milliseconds = 1000 milliseconds = 1 second
            print int(round(time.time() * 1000))
            time.sleep((delay/1000))
    elif testType == "CreateData":

        print "Create Data test finished."
    elif testType == "CreateThenDeleteData":

        print "Create then Delete Data test finished."
    elif testType == "CreateAndDeleteNodes":
        
        print "Create and Delete Lists test finished."
    elif testType == "CreateAndDeleteLists":
        cadl = CreateAndDeleteLists.initialize()
        print "Create and Delete Lists test finished."
    elif testType == "CreateAndDeleteCycles":
        
        print "Create and Delete Cycles test finished."
    else: 
        print "invalid test type input"

# Create new threads
timerThread = ScheduledService(1, "Timer", 10)
dataThread = ScheduledService(2, "CreateAndDeleteLists", 10)

# Start new Threads
timerThread.start()
timerThread.start()

print "Exiting Main Thread"
